### PR TITLE
hack/graph.sh: Add a script for rendering Cincinnati to DOT

### PR DESCRIPTION
**Pull request by wking**
Also has a drive-by fix for "Figure 2" -> "Figure 1" typo from 1221d80710 (#103).

### CODE REVIEW

**Comment by wking** [comment]
```console
$ hack/graph.sh | dot -Tpng >graph.png
```

graph

File: hack/graph.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  (
+    [
+      .edges[] |
+      "  " + (.[0] | tostring) + "->" + (.[1] | tostring) + ";"
+    ] | join("\n")
+  ) as \$edges |
+  [\$header, \$nodes, \$edges, "}"] | join("\n")
+EOF
+)"
+curl --silent --header 'Accept:application/json' "${URI}" | jq -r "${JQ_SCRIPT}"
```
Comment (steveej) [inline hack/graph.sh]: I would decrease the scope of this program and make it expect the content on stdin, leaving the fetching of the graph to the caller.

File: hack/graph.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# Usage:
+#
+#   graph.sh [URI] >graph.dot
+#
+# For example:
+#
+#   graph.sh https://cincinnati.example.org/your/graph | dot -Tsvg >graph.svg
+set -e
+
+URI="${1:-https://api.openshift.com/api/upgrades_info/v1/graph?channel=prerelease-4.1}"
```
Comment (steveej) [inline hack/graph.sh]: I was thinking of a way to make this URI a template with a nested variable for the channel as this is very guaranteed to change in the future. However that thought was superseded by the comment below, which solves the issue for this script altogether.

File: hack/graph.sh (hunk 1 above)
Comment (wking) [reply hack/graph.sh]: > I would decrease the scope of this program and make it expect the content on stdin...

I think it's a bit tedious to have to remind myself of "the usual" URI and set the `Accept` header.  But I don't feel *that* strongly about it ;).  Done with a3bb428 -> 7ec4253.

File: hack/graph.sh (hunk 2 above)
Comment (wking) [reply hack/graph.sh]: Obsolete after [this][1].

[1]: https://github.com/openshift/cincinnati/pull/108#discussion_r288158349

**Pull request merged**